### PR TITLE
Small code cleanups

### PR DESCRIPTION
**Pull request by asinghvi17**
- The ColorScheme struct documentation was getting overwritten by the function, so I just dropped that.
- Slightly more informative error message when `rangescale` is unrecognized

### CODE REVIEW

**Comment by asinghvi17** [comment]
Very interesting.  I'll revert that and look for why.

**Comment by asinghvi17** [comment]
Let's see if this works - I think it might have been a macro issue.

**Comment by asinghvi17** [comment]
I think this is one for the "cool hack" club:
```julia
julia> import Base.:/

julia> a::String/b::String = joinpath(a, b)
/ (generic function with 116 methods)

julia> 1/2
0.5

julia> @__DIR__/"a"
ERROR: syntax: "/" is not a unary operator
Stacktrace:
    Function         Module  Signature
    ────────         ──────  ─────────
[1] top-level scope
REPL[5]:0

(@__DIR__) / "a"
```

**Comment by cormullion** [comment]
perhaps `joinpath` is better here?

File: src/ColorSchemes.jl [hunk 1]
```diff
@@ -21,29 +21,26 @@ export ColorScheme,
...
 """
-ColorScheme(colors, category::AbstractString) = ColorScheme(colors, category, "")
-ColorScheme(colors) = ColorScheme(colors, "general", "")
+struct ColorScheme
+    colors::Vector{C} where {C <: Colorant}
+    category::AbstractString
+    notes::AbstractString
+    
+    ColorScheme(colors::Vector{<: Colorant}, category::AbstractString = "", notes::AbstractString = "") = new(colors, category, notes)
+end
```
Comment (asinghvi17) [inline src/ColorSchemes.jl:44]:
```suggestion

Base.getindex(cs::ColorScheme, i::AbstractFloat) = get(cs, i)
Base.getindex(cs::ColorScheme, i::AbstractVector{<: AbstractFloat}) = get(cs, i)

**Comment by asinghvi17** [comment]
Yes, I've reworked that to use `joinpath`, and tests seem to pass.  That example was just a hack which I remembered was possible.

File: src/ColorSchemes.jl (hunk 1 above)
Comment (asinghvi17) [reply src/ColorSchemes.jl:44]: @cormullion would you want this sort of a feature?  You could then do `cs[0.5]` in a colorscheme, which is a small convenience - but comes in handy sometimes.

Comment (cormullion) [reply src/ColorSchemes.jl:44]: Looks cool! Anything is good as long as it's not slower or more complicated! :)

**Comment by asinghvi17** [comment]
The documentation build is failing because of a compat constraint.  I'll add CompatHelper and adjust that.

**Comment by cormullion** [comment]
good stuff!

**Pull request merged**